plugins/modules/iosxr_bgp_global.py

```diff
@@ -0,0 +1,923 @@
+                      description: Wait indefinitely for a response
+                      type: bool
+                shutdown:
+                  type: bool
+                  description: Shutdown the RPKI cache-server
+                transport:
+                  type: dict
+                  description: Specify a transport method for the RPKI cache-server
+                  suboptions:
+                    ssh:
+                      description: Connect to the RPKI cache-server using SSH
+                      type: dict
+                      suboptions:
+                        port:
+                          description: Specify a port number for the RPKI cache-server transport
+                          type: int
+                    tcp:
+                      description: Connect to the RPKI cache-server using TCP (unencrypted)
+                      type: dict
+                      suboptions:
+                        port:
+                          description: Specify a port number for the RPKI cache-server transport
+                          type: int
+        vrfs:
+          description: Specify a vrf name.
+          type: list
+          elements: dict
+          suboptions:
+            vrf:
+              description: VRF name.
+              type: str
+            bfd: *bfd
+            bgp:
+              description: BGP commands.
+              type: dict
+              suboptions:
+                auto_policy_soft_reset: *auto_policy_soft_reset
+                bestpath: *bestpath
+                default: *default
+                enforce_first_as: *enforce_first_as
+                fast_external_fallover: *fast_external_fallover
+                log: *bgp_log
+                multipath: *multipath
+                redistribute_internal: *redistribute_internal
+                router_id: *router_id
+                unsafe_ebgp_policy: *ebgp_policy
+            default_information: *default_info
+            default_metric: *default_metric
+            mpls: *mpls
+            neighbors: *neighbors
+            rd:
+              description: route distinguisher.
+              type: dict
+              suboptions:
+                auto:
+                  description: Automatic route distinguisher.
+                  type: bool
+            socket: *socket
+            timers: *timers
+    running_config:
+      description:
+      - This option is used only with state I(parsed).
+      - The value of this option should be the output received from the IOSXR device by
+        executing the command B(show running-config router bgp).
+      - The state I(parsed) reads the configuration from C(running_config) option and
+        transforms it into Ansible structured data as per the resource module's argspec
+        and the value is then returned in the I(parsed) key within the result.
+      type: str
+    state:
+      description:
+      - The state the configuration should be left in.
```

Comment:
Can we have some details on how `state: purged` would behave? And also details on deleted/replaced w.r.t AF. You can refer https://github.com/ansible-collections/cisco.nxos/blob/main/plugins/modules/nxos_bgp_global.py#L498-L509.